src/NServiceBus.Core.Tests/ThrowsEx.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace NServiceBus.Core.Tests
+{
+    using System;
+    using NUnit.Framework;
+    using NUnit.Framework.Constraints;
+
+    static class ThrowsEx
+    {
+        public static IResolveConstraint ExceptionWithFriendlyMessage(Type instanceType, string methodName, string returnTypeName = "Task")
```

Comment:
How about making this an extension method on what is returned by `Throws.Exception`?

https://github.com/nunit/dev/wiki/Extended-Constraint-Syntax